src/services/http-client.service.ts

```diff
@@ -42,44 +42,75 @@ export class HttpClientService {
         });
     }
 
-    public async updateData(url: string, profile: Profile, body: object): Promise<any> {
+    public async updateData(url: string, headers: Headers, body: object): Promise<any> {
         return new Promise<any>((resolve, reject) => {
-            request.put(url, this.makeOptions(profile, body), (err, res) => {
+            request.put(url, this.makeOptions(headers, body), (err, res) => {
                 this.handleResponse(res, resolve, reject);
             });
         });
     }
 
-    public async findAll(url: string, profile: Profile): Promise<any> {
+    public async findAll(url: string, headers: Headers): Promise<any> {
         return new Promise<any>((resolve, reject) => {
-            request.get(url, this.makeOptions(profile), (err, res) => {
+            request.get(url, this.makeOptions(headers, null), (err, res) => {
                 this.handleResponse(res, resolve, reject);
             });
         });
     }
 
-    private makeOptions(profile: Profile, body: object = {}): CoreOptions {
+    public async buildRequestHeadersWithAuthentication(profile: Profile): Promise<any> {
```

Comment:
buildRequestHeadersWithAuthentication method is called before main method (pull/push etc.), first check if authenticationType exists. If exists, create headers and return it. Else check both 'Bearer' and 'AppKey' then return authenticated headers.

Reply:
I think we can simplify this part, so that we completely remove this logic of checking for authenticationType in the http-client, and just do it when creating the profile. So we have two cases when the authenticationType doesn't exists: 

1. Using content-cli with env variables
2. Profiles that were created before these changes don't have the authenticationType

For the first point I think it would be better to make the same change as when creating a new profile [here](https://github.com/celonis/content-cli/pull/71/files#diff-b85bcaaabf1adeb99ad93e463f998f0d61bc5e66b80052e11b38589929ba1d3fR70).
```typescript
private async buildProfileFromEnvVariables(): Promise<Profile> {
        const profileVariables = this.getProfileEnvVariables();
        const profile: Profile = {
            name: profileVariables.teamUrl,
            team: profileVariables.teamUrl,
            apiToken: profileVariables.apiToken,
            authenticationType: AuthenticationType.BEARER
        };
        profile.authenticationType = await ProfileValidator.validateProfile(profile);
        return profile;
    }
```

For the second point, you can use the fact that the previously created profiles have authenticationType `Bearer`, and substitute the `buildRequestHeadersWithAuthentication()` method with another like the following:
```typescript
private buildAuthorizationHeaders(profile: Profile): Headers {
        const authenticationType = profile.authenticationType || AuthenticationType.BEARER;
        return {
            authorization: `${authenticationType} ${apiToken}`,
            "content-type": "application/json",
        };
    }
```
That first statement in the method would make sure that if a profile doesn't have authenticationType defined, it would use the `Bearer` one by default, and solve the problem.

After these changes you wouldn't need to do this call after every command (push, pull), but just call the `buildAuthorizationHeaders` method to create the headers like before.